File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/EventHandlerMethodExecutor.cs
using System;
using System.Threading.Tasks;
using Volo.Abp.EventBus.Distributed;

namespace Volo.Abp.EventBus;

public delegate void EventHandlerMethodExecutorAsync(IEventHandler target, Object parameter);

public interface IEventHandlerMethodExecutor
{
    EventHandlerMethodExecutorAsync ExecutorAsync;//  { get; }
}

public class LocalEventHandlerMethodExecutor<TEvent> : IEventHandlerMethodExecutor
    where TEvent : class
{
    public  EventHandlerMethodExecutorAsync ExecutorAsync => (target, parameter) => target.As<ILocalEventHandler<TEvent>>().HandleEventAsync(parameter.As<TEvent>());

    public void ExecuteAsync(IEventHandler target, TEvent parameters)
    {
        return ExecutorAsync(target, parameters);
    }
}

public class DistributedEventHandlerMethodExecutor<TEvent> : IEventHandlerMethodExecutor
    where TEvent : class
{
    public EventHandlerMethodExecutorAsync ExecutorAsync => (target, parameter) => target.As<IDistributedEventHandler<TEvent>>().HandleEventAsync(parameter.As<TEvent>());

    public void ExecuteAsync(IEventHandler target, TEvent parameters)
    {
        return ExecutorAsync(target, parameters);
    }
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/EventHandlerMethodExecutor.java
package com.ueueo.eventbus;

import com.ueueo.eventbus.distributed.IDistributedEventHandler;
import com.ueueo.eventbus.local.ILocalEventHandler;

/**
 * TODO Description Of This JAVA Class.
 *
 * @author Lee
 * @date 2022-06-14 10:28
 */
public class EventHandlerMethodExecutor {

    public static class LocalEventHandlerMethodExecutor implements IEventHandlerMethodExecutor {

        @Override
        public void execute(IEventHandler target, Object parameters) {
            ((ILocalEventHandler) target).handleEvent(parameters);
        }
    }

    public static class DistributedEventHandlerMethodExecutor implements IEventHandlerMethodExecutor {
        @Override
        public void execute(IEventHandler target, Object parameters) {
            ((IDistributedEventHandler) target).handleEvent(parameters);
        }
    }

}
